build/Makefile

```diff
@@ -391,7 +391,6 @@ install: LOG_LEVEL := "debug"
 install: EXTERNAL_IP ?= $(shell $(DOCKER_RUN) kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
 install: FEATURE_GATES ?= $(ALPHA_FEATURE_GATES)
 install: SERVICE_MONITOR := false
-install: CONTROLLER_SAFE_TO_EVICT := false # TODO(#2797): Remove when `SplitControllerAndExtensions` => safe-to-evict=true
```

Comment:
Should this be left in until `SplitControllerAndExtensions` is enabled by default (e.g. beta)?

Reply:
I think since we have a check within the yaml, we won't need this. But in the case that `safe-to-evict` is enabled, and we still want it to be false for the `controller` we would want this. Not sure when that will happen though. What do you think?

Reply:
This only really matters for e2e testing by CI. If you look at how we've configured CI for Autopilot, it's a bit asymmetric now, with the `SaveToEvict` and `SplitControllerAndExtensions` feature gates being enabled in both the `-stable` and `-feature-gates` runs - basically these feature gates are mandatory on Autopilot. I added `CONTROLLER_SAFE_TO_EVICT` last week to allow us to add Autopilot to CI without gating on @chiayi's work, but now that we have the feature gate, I would rather drop the option. Dropping the option also means that on `e2e-stable` with the default gates on GKE returns to the normal configuration, rather than as it is today, where it's testing a slightly off config (stable gates + safe-to-evict annotation).

Reply:
Thanks for the explanation. Make sense!.